workshops/digit_recognition/KerasMNIST.md

```diff
@@ -0,0 +1,527 @@
+![overview of different stages/phases of tutorial](https://cloud-h61jcon63.vercel.app/0image.png)
+
+I'd recommend opening this tutorial on [Google Colab](https://colab.research.google.com/drive/1S0rZ4htWmO-S7npfMvNyMZrDpt8XCaW9) and then make a [new](https://colab.research.google.com/#create=true) Colab to get started!
+
+### Importing Tensorflow and Dependencies
+
+Let's start off by importing all the required dependencies. We will be using [TensorFlow](https://tensorflow.org), an open-source machine learning platform. Within TensorFlow we will use  the [Keras Sequential API](https://www.tensorflow.org/guide/keras/overview), which provides a layer of abstraction on top of all the underlying math theory (though if you are mathematically-inclined, I'd totally recommend diving into the math. Andrew Ng's [Coursera](https://www.coursera.org/learn/machine-learning) course is a great starting place). Using Keras will greatly simplify the process, allowing us to create and train our model with ease.
+
+
+```py
+#for rendering pictures and labels properly
+%matplotlib inline 
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+try:
+  # %tensorflow_version only exists in Colab.
+  %tensorflow_version 2.x
+except Exception:
+  pass
+import tensorflow as tf
+```
+
+### Load Training Data
+
+For any machine learning (and particularly deep learning) project, we need data. In our case, we need access to a dataset of thousands of handwritten digits. Lucky for us, [MNIST Images](http://yann.lecun.com/exdb/mnist/) does just that. The MNIST dataset was taken from the writing of US Census Bureau officials and high school students, and is commonly used to test/train simple machine learning models. Let's go ahead and load in the dataset using Keras.
+
+
+```py
+nb_classes = 10
+
+# the data, shuffled and split between tran and test sets
+(X_train, Y_train), (X_test, Y_test) = tf.keras.datasets.mnist.load_data()
+print("X_train original shape", X_train.shape)
+print("y_train original shape", Y_train.shape)
+```
+
+    Downloading data from https://storage.googleapis.com/tensorflow/tf-keras-datasets/mnist.npz
+    11493376/11490434 [==============================] - 0s 0us/step
+    X_train original shape (60000, 28, 28)
+    y_train original shape (60000,)
+
+
+Before preparing the dataset, let us look at some samples of training data. When approaching machine learning problems, visualizing data, when possible, is a great thing to do since it gives useful insights.
+
+
+```py
+for i in range(9):
+    plt.subplot(3,3,i+1)
+    plt.imshow(X_train[i], cmap='gray', interpolation='none')
+    plt.title("Class {}".format(Y_train[i]))
+#to make sure the spacing between the plots is proper
+plt.tight_layout()
+```
+
+
+![examples of correct predictions](https://cloud-lc775uvcv.vercel.app/3kerasmnist_9_0.png)
+
+
+
+```py
+print(X_train[0])
+print("Shape: " + str(X_train[0].shape))
+```
+
+    [[  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
```

Comment:
can you please tell something about what this is? (like a one line description of this block)